core/interfaces/tags.py

```diff
@@ -25,24 +25,39 @@
 END_PUSH_MESSAGE = "Push complete."
 
 
-def smart_incremented_tag(tag, release):
+def smart_incremented_tag(tag, release_type):
     """
     Automatic increment of a given tag depending on the type of release.
     """
-    r = re.compile(r"^([0-9A-Za-z-]*[A-Za-z-])*?([0-9]+)\.([0-9]+)\.([0-9]+)(.*?)$")
-    m = r.match(tag)
-    if m:
-        prefix, major, minor, patch, _ = m.groups()
-        prefix = "" if not prefix else prefix
-        if release == "major":
-            major = str(int(major)+1)
-            minor = patch = "0"
-        elif release == "minor":
-            minor = str(int(minor)+1)
-            patch = "0"
-        elif release == "patch":
-            patch = str(int(patch)+1)
-        return prefix + major + "." + minor + "." + patch
+    if release_type in ["major", "minor", "patch"]:
+        r = re.compile(r"^([0-9A-Za-z-]*[A-Za-z-])?([0-9]+)\.([0-9]+)\.([0-9]+)$")
```

Comment:
Please move these compiled regular expressions to the top of the file, with a descriptive name.  No need to compile each time the command is run, just on start-up.